### PR TITLE
Fix issue 673 trailing spaces of cell values lost

### PR DESCRIPTION
**Pull request by sandraros**

### CODE REVIEW

**Comment by sandraros** [comment]
@larshp This fix is rejected because of lint check:
> Statement does not exist in ABAPv702(or a parser error), "IF".

Concerned code is:
```
      IF contains( val = <fs_sheet_string>-string_value start = ` ` )
            OR contains( val = <fs_sheet_string>-string_value end = ` ` ).
```
I guess that "contains" predicate function was added in 7.02? Or am I wrong? Thanks.

File: src/zcl_excel_writer_2007.clas.abap [hunk 1]
```diff
@@ -3593,8 +3593,8 @@ CLASS zcl_excel_writer_2007 IMPLEMENTATION.
                                                        parent = lo_document ).
       lo_sub_element = lo_document->create_simple_element( name   = lc_xml_node_t
                                                            parent = lo_document ).
-*    if <fs_sheet_string>-string_type EQ 's_leading_blanks'.
-      IF <fs_sheet_string>-string_value IS NOT INITIAL AND <fs_sheet_string>-string_value(1) EQ ` `.
+      IF contains( val = <fs_sheet_string>-string_value start = ` ` )
+            OR contains( val = <fs_sheet_string>-string_value end = ` ` ).
```
Comment (larshp) [inline src/zcl_excel_writer_2007.clas.abap]:
```suggestion
      IF contains( val = <fs_sheet_string>-string_value start = ` ` ) = abap_true
            OR contains( val = <fs_sheet_string>-string_value end = ` ` ) = abap_true.
```

**Review by larshp — APPROVED** [review]
looks good to me 👍

**Comment by sandraros** [comment]
@larshp Urgent. Coming back to adding `= abap_true` to `IF contains( ... )` (predicate function), I get a **short dump/syntax error** in my 7.52SP0/7.52SP1 systems (optionally could be solved by wrapping with `boolc` (`IF boolc( contains( ... ) ) = abap_true ).`:
image
or better, lint to be fixed, or was there a reason to add `= abap_true`?

@AndreaBorgia-Abo Could you check if you have a syntax error with the writer of latest master?

**Comment by AndreaBorgia-Abo** [comment]
Yup, HANA1909, same error.

**Comment by sandraros** [comment]
@AndreaBorgia-Abo Could you check the corrective PR #812 please? (it sounds just stupid to have overcomplex statement `IF boolc( contains( ... ) ) = abap_true` for now, but works at least/until confirmation by @larshp)

**Comment by sandraros** [comment]
Thanks @AndreaBorgia-Abo 😀 (I'm feeling less guilty 😓)

**Pull request merged**